### PR TITLE
[RFR] Db migrate

### PR DESCRIPTION
**Pull request by psav**
Migration of DB to new object

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0cf86db3c5416dd8196f5dd4716e926161f8487c:

`cfme/common/__init__.py`:
- :bangbang: Line 142:101: **E501** *line too long (103 > 100 characters)*

`cfme/middleware/server_group.py`:
- :bangbang: Line 25:101: **E501** *line too long (106 > 100 characters)*

`cfme/tests/intelligence/reports/test_validate_chargeback_report.py`:
- :bangbang: Line 147:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 173:101: **E501** *line too long (101 > 100 characters)*
- :bangbang: Line 208:101: **E501** *line too long (106 > 100 characters)*

`cfme/tests/test_utilization_metrics.py`:
- :bangbang: Line 129:30: **E999** *SyntaxError: invalid syntax*
- :bangbang: Line 145:101: **E501** *line too long (102 > 100 characters)*

`utils/events.py`:
- :bangbang: Line 90:101: **E501** *line too long (104 > 100 characters)*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 16601cef5103db188279865cabce887c5b57e1ab:

`cfme/common/__init__.py`:
- :bangbang: Line 142:101: **E501** *line too long (103 > 100 characters)*

`cfme/middleware/server_group.py`:
- :bangbang: Line 25:101: **E501** *line too long (106 > 100 characters)*

`cfme/tests/test_utilization_metrics.py`:
- :bangbang: Line 129:30: **E999** *SyntaxError: invalid syntax*
- :bangbang: Line 145:101: **E501** *line too long (102 > 100 characters)*

`utils/events.py`:
- :bangbang: Line 90:101: **E501** *line too long (104 > 100 characters)*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: utils/appliance/db.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+            self.log.error('Unable to pull database address from appliance')
+            self.log.exception(exc)
+            return self.appliance.address
+
+    @property
+    def is_partition_extended(self):
+        return self.appliance.ssh_client.run_command(
+            "ls /var/www/miq/vmdb/.db_partition_extended") == 0
+
+    @logger_wrap("Extend DB partition")
```
Comment (mfalesni) [inline utils/appliance/db.py]: wouldn't be the usage of self.logger better?

Comment (psav) [inline utils/appliance/db.py]: I thoguht about it, but a) at the moment they are both using the same one, b) I like the logger wrapper that gives the prefix on all log messages in that unit

Comment (psav) [inline utils/appliance/db.py]: However, I just had a thought, what about we use self.log and make the logger wrapper a context manager that prefixes all the calls?

File: utils/appliance/db.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+        wait_for(_db_dropped, delay=5, timeout=60, message="drop the vmdb_production DB")
+
+    @property
+    def postgres_version(self):
+        # postgres's version is in the service name and file paths when we pull it from SCL,
+        # so this is a little resolver to help keep the version picking centralized
+        return 'rh-postgresql95'
+
+    @property
+    def ssh_client(self, **connect_kwargs):
```
Comment (mfalesni) [inline utils/appliance/db.py]: it is a property, you can nuke the kwargs here.

Comment (psav) [inline utils/appliance/db.py]: Right, it was a straight copy

File: cfme/tests/test_db_restore.py [hunk 3]
```diff
@@ -105,8 +105,8 @@ def finalize():
     with appl2.ipapp:
         # Restore DB on the second appliance
         appl2.ipapp.evmserverd.stop()
-        appl2.ipapp.drop_database()
-        appl2.ipapp.restore_database()
+        appl2.ipapp.db.drop()
+        appl2.ipapp.restore()
```
Comment (mfalesni) [inline cfme/tests/test_db_restore.py]: `.db.restore()` ?

**Comment by mfalesni** [comment]
@psav looks good to me, I had one question there, if that is typo or not :)

**Pull request merged**